onnx/reference/ops/op_dequantize_linear.py

```diff
@@ -54,26 +54,31 @@ def _run(
             raise RuntimeError("Input 2 must be a vector or a number.")
 
         x_type = self.get_x_type(x)
-        if x_zero_point is not None:
+        f8_type = x_type in {
+            TensorProto.FLOAT8E4M3FN,
+            TensorProto.FLOAT8E4M3FNUZ,
+            TensorProto.FLOAT8E5M2,
+            TensorProto.FLOAT8E5M2FNUZ,
+        }
+        if x_zero_point is not None and not f8_type:
             zero_type = self.get_x_type(x_zero_point)
             if x_type != zero_type:
                 raise RuntimeError(
                     f"Type mismatch {x_type} != {zero_type} in DequantizeLinear."
                 )
-            if x_type in (
-                TensorProto.FLOAT8E4M3FN,
-                TensorProto.FLOAT8E4M3FNUZ,
-                TensorProto.FLOAT8E5M2,
-                TensorProto.FLOAT8E5M2FNUZ,
-            ):
-                raise RuntimeError(
-                    f"x_zero_point not supported for float 8 type {x_type}."
-                )
 
             dx = x.astype(np.float32) - DequantizeLinear.reshape_input(
                 x_zero_point, x.shape, axis
             )
         else:
+            if (
+                f8_type
+                and x_zero_point is not None
+                and x_zero_point.astype(np.uint8) != np.uint8(0)
+            ):
+                raise RuntimeError(
+                    "x_zero_point is not null but should be for float 8 types."
```

Comment:
nit: typo? `should be ...`

Reply:
I read it wrong sorry.

Reply:
> Do you mean "x_zero_point is not null but should be null for float 8 types."?

Yeah that would make it clearer just from a readability's perspective

Reply:
done